controlplane/kubeadm/internal/controllers/remediation.go

```diff
@@ -168,6 +169,23 @@ func (r *KubeadmControlPlaneReconciler) reconcileUnhealthyMachines(ctx context.C
 	// Before starting remediation, run preflight checks in order to verify it is safe to remediate.
 	// If any of the following checks fails, we'll surface the reason in the MachineOwnerRemediated condition.
 
+	if feature.Gates.Enabled(feature.ClusterTopology) {
+		// Skip remediation when we expect an upgrade to be propagated from the cluster topology.
+		if controlPlane.Cluster.Spec.Topology != nil && controlPlane.Cluster.Spec.Topology.Version != controlPlane.KCP.Spec.Version {
+			log.Info("A control plane machine needs remediation, but waiting for a pending version upgrade. Skipping remediation", "version", controlPlane.Cluster.Spec.Topology.Version)
```

Comment:
I don't know if folks who read this will know what we mean with "pending version upgrade". Also wonder if they know what the version value is. I would include the version value in the message and clearly say that we wait for the version from Cluster.spec.topology.version to be propagated to KubeadmControlPlane.spec.version

(same for the conditions below)